lib/src/main/java/io/ably/lib/http/AsyncHttp.java

```diff
@@ -147,6 +148,8 @@ private AblyRequestWithFallback(
 		public void run() {
 			int retryCountRemaining = Hosts.isRestFallbackSupported(http.host) ? http.options.httpMaxRetryCount : 0;
 			String candidateHost = http.host;
+			if(http.options.fallbackHosts != null)
+				Collections.shuffle(http.options.fallbackHosts);
```

Comment:
Perhaps we should raise an issue and defer fixing this, but @paddybyers was concerned with shuffling unnecessarily for every HTTP request instead of just shuffling once at the outset the the custom fallbackhosts are provided.